server/issue_handler.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/mattermost/mattermost-mattermod/model"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+func (s *Server) issueEventHandler(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
+	defer cancel()
+
+	event, err := PullRequestEventFromJSON(r.Body)
```

Comment:
I am a bit confused here. I am assuming we are migrating the issues webhook here. Shouldn't we use the `EventDataFromJSON` function here? Even that does not match the exact format [here]( https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#issues).

I would just have 3 separate `xxFromJSON` functions for 3 separate events. It's much clearer this way.
- IssueFromJSON
- IssueCommentFromJSON
- PullRequestFromJSON

Reply:
You're right, this is confusing. Actually I wanted to migrate gradually to not break things since we are shipping the code immediately :) However, we can have a new `type` for this handler which only includes `action` and `issue` itself. In fact, we can handle this in parsing stage and use our `model.Issue` for the rest of the code. Will update this accordingly.